modules/currency-l0/src/main/scala/org/tessellation/currency/l0/snapshot/programs/Rollback.scala

```diff
@@ -70,7 +71,20 @@ object Rollback {
 
       _ <- dataApplication.map {
         case ((da, cs)) =>
-          val fetchSnapshot: Hash => F[Option[Hashed[GlobalIncrementalSnapshot]]] = (hash: Hash) => globalL0Service.pullGlobalSnapshot(hash)
+          val attempts_to_fetch_global_snapshot = 10
+          val fetchSnapshot: Hash => F[Option[Hashed[GlobalIncrementalSnapshot]]] = (hash: Hash) => {
+            def attemptFetch(remainingAttempts: Int): F[Option[Hashed[GlobalIncrementalSnapshot]]] =
+              globalL0Service.pullGlobalSnapshot(hash).flatMap {
+                case Some(snapshot) => snapshot.some.pure
+                case None if remainingAttempts > 1 =>
```

Comment:
is `None` returned only when peer explicitly says there is no such snapshot, or is `None` returned only when there was an error (like the node was unresponsive or there were some networking issues)?